### ISSUE TITLE
Please add Protocol bit-structure support for Network diagram in PlantUML

### ISSUE THREAD

**Issue by liuhangbin** [plantuml/plantuml#1523]
**Is your feature request related to a problem? Please describe.**
The Protocol bit-structure is always used to describe what a packet structure looks like.  But I can't find a similar example in plantuml nwdiag. https://plantuml.com/nwdiag . Would you please help add this support?

**Describe the solution you'd like**
Here are what other tools look like

http://blockdiag.com/en/nwdiag/packetdiag-examples.html.
https://kroki.io/examples.html#packetdiag

**Describe alternatives you've considered**

**Additional context**

It also would be good if we could set the values for each field.  see the picture as an example.

![1](https://github.com/plantuml/plantuml/assets/1594572/62df8ba1-5db5-42ad-ad99-ab34746350cc)

**Comment by liuhangbin** [comment]
Here is another example of the [bytefield](https://kroki.io/examples.html#bytefield)

**Comment by The-Lum** [comment]
_[Just for traceability] See also similar request here:_
- https://forum.plantuml.net/8787/nwdiag-package-includes-scripts-called-rackdiag-packetdiag
- https://forum.plantuml.net/9841/protocol-bit-structure-tool
- https://forum.plantuml.net/16484/can-use-plantuml-for-describing-the-octet-layout-of-message

Regards,
Th.

**Comment by liuhangbin** [comment]
> _[Just for traceability] See also similar request here:_
> 
> * https://forum.plantuml.net/8787/nwdiag-package-includes-scripts-called-rackdiag-packetdiag
> * https://forum.plantuml.net/9841/protocol-bit-structure-tool
> * https://forum.plantuml.net/16484/can-use-plantuml-for-describing-the-octet-layout-of-message

Yes, I also saw the requests in the forum. But I don't see progress on the forum. So I opened this request.

**Comment by kolulu23** [comment]
I just started to work on this, adding both [rackdiag](http://blockdiag.com/en/nwdiag/rackdiag-examples.html) and [packetdiag](http://blockdiag.com/en/nwdiag/packetdiag-examples.html) from `nwdiag` into PlantUML.

However, I do think these two diagrams provide little features compared to other newly added diagram formats. For protocol protraiting diagrams, something like [bytefield](https://texdoc.org/serve/bytefield.pdf/0) seems to be more appealing.
As of now, use [Ditta diagram](https://plantuml.com/ditaa) to generate packet diagram like [this](https://editor.plantuml.com/uml/hPHFImCn4CNl-odUiQLKjDQ_KAWi8eAreLsAr-mcre7kR4soXqe-l2bQ3xMswkRD8J2GoM_U31bQ9urJvAHCZd627aut9_eumpakS8ah_26NTBeD8_6pGSyB_-I1D8KVUOWCd_Am-hNMN6XmNURQuBl2lHWKS5JgO52vtF76OVau47R5nOkfcDwaeufXgzg4vblJwIHjWvHcHy-qquSrHD8PD8p0K5Ji3EMr8vv_0QHR0FP78qA5ew1_kHYnYsOAu_GOYTkxRXypihLSm3gZPObMld8QSa5prgfzg4G4YltwpLBjrtKOrbXaz-F7cmTCCe6bL6ez-Wr4XCKUQbOV_yux) might fit your need but it has its own limitations.

<img width="710" height="308" alt="Image" src="https://github.com/user-attachments/assets/2251604e-7c9e-42a5-8337-54e443ee8def" />

That being said, adding `rackdiag` and `packetdiag` to PlantUML to complete current `nwdiag` is still reasonable. 

@The-Lum @arnaudroques What do you think

**Comment by The-Lum** [comment]
Hello @kolulu23

> That being said, adding `rackdiag` and `packetdiag` to PlantUML to complete current `nwdiag` is still reasonable.
>  
> - What do you think
 
Well, here are some first questions :
- Where is the source code of those tools?
- Is it on the right license?

Because FYI, `nwdiag` is now developed on python... see:
- https://github.com/blockdiag/nwdiag/

And that seems the same for `packetdiag` & `rackdiag`:
- https://github.com/blockdiag/nwdiag/tree/master/src/packetdiag
- https://github.com/blockdiag/nwdiag/tree/master/src/rackdiag

Next... ⏭️ 
Then we now wait contribution... and contributors... to do the expected things on Java.

> [!TIP]
>  
>Call for contributions
>- Who wants to port Python code to Java code?

Regards,
Th.

**Comment by kolulu23** [comment]
@The-Lum Sorry for the late reply, I made an `packetdiag` implementation that is relatively close to the original. Could you review my code when you have time? https://github.com/plantuml/plantuml/pull/2520

Example diag:
```
@startpacketdiag
{
  colwidth = 32
  node_height = 72

  0-15: Source Port
  16-31: Destination Port
  32-63: Sequence Number
  64-95: Acknowledgment Number
  96-99: Data Offset
  100-105: Reserved
  106: URG 
  107: ACK 
  108: PSH 
  109: RST 
  110: SYN 
  111: FIN
  112-127: Window
  128-143: Checksum
  144-159: Urgent Pointer
  160-191: (Options and Padding)
  192-223: data [colheight = 3]
}
@endpacketdiag
```

<img width="1373" height="572" alt="Image" src="https://github.com/user-attachments/assets/bea00bf2-538e-4382-859b-511766075720" />

this is my first try on implementing features for plantuml so there might be some rough edges in my code
